rskj-core/src/main/java/co/rsk/net/discovery/PeerExplorer.java

```diff
@@ -259,20 +261,29 @@ public NeighborsPeerMessage sendNeighbors(InetSocketAddress nodeAddress, List<No
     }
 
     public void purgeRequests() {
-        List<PeerDiscoveryRequest> oldPingRequests = this.removeExpiredRequests(this.pendingPingRequests);
-        this.resendExpiredPing(oldPingRequests);
-        this.removeConnections(oldPingRequests.stream().
-                filter(r -> r.getAttemptNumber() >= 3).collect(Collectors.toList()));
+        List<PeerDiscoveryRequest> oldPingRequests = removeExpiredRequests(this.pendingPingRequests);
+        removeExpiredChallenges(oldPingRequests);
+        resendExpiredPing(oldPingRequests);
+        removeConnections(oldPingRequests.stream().
+                filter(r -> r.getAttemptNumber() >= RETRIES_COUNT).collect(Collectors.toList()));
 
         removeExpiredRequests(this.pendingFindNodeRequests);
     }
 
+    @VisibleForTesting
     public void cleanAndUpdate() {
```

Comment:
`public` but only used by tests now.

Reply:
Now this is only used in the `co.rsk.net.discovery.UDPServerTest#run3NodesFullTest`, I would remove it and replace it with the explicit call for both methods (it's only 6 usages)

Reply:
Removed :+1: